docs/features/incremental-generators.cookbook.md

```diff
@@ -0,0 +1,656 @@
+
+        context.RegisterSourceOutput(pipeline,
+            static (context, pair) => 
+                // Note: this AddSource is simplified. You will likely want to include the path in the name of the file to avoid
+                // issues with duplicate file names in different paths in the same project.
+                context.AddSource($"{pair.Name}generated.cs", SourceText.From(pair.Code, Encoding.UTF8)));
+    }
+}
+```
+
+### Augment user code
+
+**User scenario:** As a generator author I want to be able to inspect and augment a user's code with new functionality.
+
+**Solution:** Require the user to make the class you want to augment be a `partial class`, and mark it with a unique attribute.
+Provide that attribute in a `RegisterPostInitializationOutput` step. Register for callbacks on that attribute with
+`ForAttributeWithMetadataName` to collect the information needed to generate code, and use tuples (or create an equatable model)
+ to pass along that information. That information should be extracted from syntax and symbols; **do not put syntax or symbols into
+ your models**.
+
+**Example:**
+
+```csharp
+public partial class UserClass
+{
+    [Generate]
+    public partial void UserMethod();
+}
+```
+
+```csharp
+[Generator]
+public class AugmentingGenerator : IIncrementalGenerator
+{
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        context.RegisterPostInitializationOutput(static postInitializationContext =>
+            postInitializationContext.AddSource("myGeneratedFile.cs", SourceText.From("""
+                using System;
+                namespace GeneratedNamespace
+                {
+                    [AttributeUsage(AttributeTargets.Method)]
+                    internal sealed class GeneratedAttribute : Attribute
+                    {
+                    }
+                }
+                """, Encoding.UTF8));
+
+        var pipeline = context.SyntaxProvider.ForAttributeWithMetadataName(
+            fullyQualifiedMetadataName: "GeneratedNamespace.GeneratedAttribute",
+            predicate: static (syntaxNode, cancellationToken) => syntaxNode is BaseMethodDeclarationSyntax,
+            transform: static (context, cancellationToken) =>
+            {
+                var containingClass = context.TargetSymbol.ContainingType;
+                return new Model(
+                    // Note: this is a simplified example. You will also need to handle the case where the type is in a global namespace, nested, etc.
+                    Namespace: containingClass.ContainingNamespace?.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted)),
+                    ClassName: containingClass.Name,
+                    MethodName: context.TargetSymbol.Name);
+            }
+        );
+
+        // Register a factory that can create our custom syntax receiver
```

Comment:
I would actually just drop this comment, it's kind of confusing. Why is it a factory?

Reply:
I don't know why I kept that comment at all.